### ISSUE TITLE
Which one is `y_true` when I use a custom loss for regression mode.

### ISSUE THREAD

**Issue by EeyoreLee** [dmlc/xgboost#10109]
The xgboost version I used is the last version. 
I browsed through the historical issues. seems the custom loss should be like
```
def custom_loss(y_pred, y_true):
    ...
    return grad, hess
```
But, I print the value of y_true on the step 0, it's all 0.5, so I think there's a update or something wrong that the custom loss should be like
```
def custom_loss(y_true, y_pred):
    ...
    return grad, hess
```

**Comment by trivialfis** [comment]
If you are using the sklearn interface, it should look like this https://github.com/dmlc/xgboost/blob/1450aebb74fe4ae5ed28913910cb425eb348166c/python-package/xgboost/testing/__init__.py#L762

If you are using the native interface, see https://github.com/dmlc/xgboost/blob/master/demo/guide-python/custom_rmsle.py

**Comment by EeyoreLee** [comment]
Thanks, that makes sense.

**Comment by trivialfis** [comment]
Apologies, the link for rmse is incorrect, this one is the custom softprob: https://github.com/dmlc/xgboost/blob/1450aebb74fe4ae5ed28913910cb425eb348166c/python-package/xgboost/testing/__init__.py#L792
